napari/resources/test_resources.py

```diff
@@ -1,9 +1,36 @@
 import os
+import platform
 
-from napari.resources import import_resources
+import qtpy
+
+from napari.resources import build_pyqt_resources, import_resources
 
 
 def test_resources():
     """Test that we can build icons and resources."""
     out = import_resources(version='test')
     os.remove(out)
+
+
+def test_path_pollution(tmp_path, monkeypatch, capfd):
+    if qtpy.API == "pyqt5":
```

Comment:
it's not entirely clear to me what cases you're catching with this test.  For instance, this passes for me (locally) already on master, without this PR.  Can you add a docstring to this test explaining the purpose of it?

Reply:
Did you use pyqt5 as backend?

Reply:
yeah

Reply:
So this test, in old version properly, work on pyside2 environment.
Current version should fail on master. 

Looks better now?